modules/account-lib/src/coin/dot/iface.ts

```diff
@@ -0,0 +1,171 @@
+import { BaseTxInfo, TypeRegistry, DecodedUnsignedTx } from '@substrate/txwrapper-core/lib/types';
+
+export interface Seed {
+  seed: Uint8Array;
+}
+
+/**
+ * Specification name type of the chain. Used in setting up the registry
+ */
+export type specNameType = 'kusama' | 'polkadot' | 'westend' | 'statemint' | 'statemine';
+
+/**
+ * Method names for the transaction method. Names change based on the type of transaction e.g 'bond' for the staking transaction
+ */
+export enum MethodNames {
+  AddProxy = 'addProxy',
+  Proxy = 'proxy',
+  Bond = 'bond',
+  TransferKeepAlive = 'transferKeepAlive',
+  Unbond = 'unbond',
+}
+
+/**
+ * Block validity interface based on the BitgoJS standard doc.
+ */
+export interface Validity {
+  firstValid: number;
+  lastValid?: number;
+}
+
+/**
+ * The transaction data returned from the toJson() function of a transaction
+ */
+export interface TxData {
```

Comment:
Is this the protocol format? Or is it custom?

Reply:
Custom JSON